fault-proof/src/proposer.rs

```diff
@@ -171,28 +171,48 @@ where
         };
 
         tracing::info!("Generating Range Proof");
-        let range_proof = if self.config.mock_mode {
+        let (range_proof, total_instruction_cycles, total_sp1_gas) = if self.config.mock_mode {
             tracing::info!("Using mock mode for range proof generation");
-            let (public_values, _) =
+            let (public_values, report) =
                 self.prover.network_prover.execute(get_range_elf_embedded(), &sp1_stdin).run()?;
```

Comment:
can we just add a note here that gas is counted by default? or do 
```suggestion
              self.prover.network_prover.execute(get_range_elf_embedded(), &sp1_stdin).calculate_gas(true).run()?;
```